lib/nostrum/struct/permission.ex

```diff
@@ -0,0 +1,115 @@
+defmodule Nostrum.Struct.Permission do
+  use Bitwise
+
+  @type general_permission ::
+          :create_instant_invite
+          | :kick_members
+          | :ban_members
+          | :administrator
+          | :manage_channels
+          | :manage_guild
+          | :view_audit_log
+          | :view_channel
+          | :change_nickname
+          | :manage_nicknames
+          | :manage_roles
+          | :manage_webhooks
+          | :manage_emojis
+
+  @type text_permission ::
+          :add_reactions
+          | :send_messages
+          | :send_tts_messages
+          | :manage_messages
+          | :embed_links
+          | :attach_files
+          | :read_message_history
+          | :mention_everyone
+          | :use_external_emojis
+
+  @type voice_permission ::
+          :connect
+          | :speak
+          | :mute_members
+          | :deafen_members
+          | :move_members
+          | :use_vad
+
+  @type t ::
+          general_permission
+          | text_permission
+          | voice_permission
+
+  @type permission_set :: MapSet.t(t)
```

Comment:
I don't think this is correct, but maybe I'm not understanding something. We're saying that it'll be a `MapSet` of `t`, but wouldn't this be able to include all three of the types mentioned above (general, text, and voice)?

If I'm correct, maybe it won't be worth separating these permission atoms into their own fields?

Also some things like `:view_channel` looks like it applies to both text and voice channels.

Reply:
Permissions are collective I think. A set of permissions can contain all the permissions. I made separate general, text, and voice permission types to organize the module like the discord client. However, I found out later that there wasn't much reason to do so :\

Reason being that I'm going to make a type for each permission. This would give documentation to each permission and makes it easy to read. It also follows how we've been documenting our modules for some time. Plus, having a type for each possible permission would allow us to link to the permission in the rest of the documentation.

Reply:
Speaking of which, would a MapSet be ideal? The MapSet has some nice functions that work well for what the user would need to compare permissions (`member?`, `difference/2`, `disjoint?/2`, `subset?/2`). However, the user would need to create a new MapSet for each time they want to make a permission list. Might be good to make a `new_set/1` then?

Let me know your thoughts.